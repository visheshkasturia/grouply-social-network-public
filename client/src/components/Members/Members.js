import React, { useState, useEffect } from 'react';
import MemberCard from '../MemberCard/MemberCard';
/* eslint-disable */
function Members({ user, group }) {
  const [groupList, setGroupList] = useState(group.allMembers)

  const getChangedMemberList = (data) => {
    setGroupList(data);
  } 

  useEffect(() => {
  }, [groupList])
  return (
    <div>
      {
        groupList.map((member, i) => (
          <MemberCard key={member} getChangedMemberList={getChangedMemberList} member={member} user={user} i={i + 1} group={group} />
        ))
      }
    </div>
  )
}

export default Members;
