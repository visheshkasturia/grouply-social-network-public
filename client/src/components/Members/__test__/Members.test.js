import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Members from '../Members'

const admin = 'abc'
const member = 'test1'
const post = '1'
const tag = 'tag'
const group = {
  groupName: "test",
  description: "desc",
  admins: [admin],
  allMembers: [member],
  allPosts: [post],
  tags: [tag]
}

it('Members renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Members group={group} />, div);
});

it('Members matches snapshot', () => {
  const tree = renderer.create(<Members group={group} />).toJSON();
  expect(tree).toMatchSnapshot();
});
