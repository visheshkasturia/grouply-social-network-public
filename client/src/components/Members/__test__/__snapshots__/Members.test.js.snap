// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members matches snapshot 1`] = `
<div>
  <div
    className="member-card-container"
  >
    <div
      className="member-card-SNo flex-center"
    >
       
      1
       
    </div>
    <div
      className="member-card-name flex-center"
    >
       
      test1
       
    </div>
    <div
      className="member-card-addAdmin flex-center"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Make Admin
      </button>
    </div>
    <div
      className="member-card-removeMember flex-center"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Remove Member
      </button>
    </div>
    <div
      className="add-admin-status"
    />
  </div>
</div>
`;
